src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -102,17 +64,118 @@
  * To configure the check:
  * </p>
  * <pre>
- * &lt;module name=&quot;FallThrough&quot;/&gt;
+ * &lt;module name="FallThrough"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * switch (i) {
+ *   case 1:
+ *     i++;
+ *   case 2: // violation, case 1 contains code but lacks break, return, throw or continue statement
+ *     i++;
+ *     break;
+ *   case 3: // OK
+ *     i++;
+ *     return;
+ *   case 4: // OK
+ *     i++;
+ *     throw new Exception();
+ *   case 5: // OK
+ *     i++;
+ *     continue;
+ *   case 6: // OK
+ *   case 7: // OK, case 6 does not contain code
+ *     i++;
+ *   // OK, by default the last case is not checked
+ * }
+ * </pre>
+ * <p>
+ * Unreachable code still causes violation:
+ * </p>
+ * <pre>
+ * switch (i) {
+ *   default:
+ *     return;
+ *   case 1:
+ *     i++; // unreachable
+ *   case 2: // violation
+ *     return;
+ * }
  * </pre>
  * <p>
- * or
+ * Add special comments to suppress the warning:
+ * </p>
+ * <pre>
+ * switch (i) {
+ *   case 1:
+ *     i++; // fall through
+ *
+ *   case 2: // OK
+ *     i++;
+ *     // fallthru
+ *   case 3: { // OK
+ *     i++;
```

Comment:
there is no comment, why it is ok?

Reply:
There is a comment `// fall through` in case 1, `// fallthru` in case 2, and `/* fall-through */` in case 3.

Reply:
then it is not clear about case 4 in this example. There is a comment, but it is outside case block

Reply:
Do you mean this
```
    case 3: {
        i++;
    }
        /* fall-through*/
    case ...
```
It is ok because the comment doesn't have to be in the case block. It must be `on the last non-empty line before the case triggering the warning`.
I ran checkstyle and there is no violation here.

Reply:
ok